File: backend/App/Domain/ValueObjects/RoleName.php
<?php

declare(strict_types=1);

namespace App\Domain\ValueObjects;

use App\Domain\ValueObjects\Base\EnumField;

enum Role: string
{
    case Client = 'client';
    case Barber = 'barber';
    case Assistant = 'assistant';
    case Admin = 'admin';
}

final readonly class RoleName extends EnumField
{
    protected const string ENUM_CLASS = Role::class;
}

File: tests/domain/value_objects/role_name.py
from dataclasses import dataclass
from enum import StrEnum

from domain.value_objects.base import EnumField


class Role(StrEnum):
    CLIENT = "client"
    BARBER = "barber"
    ASSISTANT = "assistant"
    ADMIN = "admin"


@dataclass(slots=True, frozen=True)
class RoleName(EnumField):
    _enum_values = tuple(Role)
